llvm/test/CodeGen/X86/x86-win64-shrink-wrapping.ll

```diff
@@ -178,12 +188,16 @@ define i32 @loopInfoSaveOutsideLoop2(i32 %cond, i32 %N) #0 {
 ; DISABLE-NEXT:    nop
 ; DISABLE-NEXT:    #NO_APP
 ; DISABLE-NEXT:    shll $3, %eax
-; DISABLE-NEXT:    jmp .LBB1_5
-; DISABLE-NEXT:  .LBB1_4: # %if.else
+; DISABLE-NEXT:    .seh_startepilogue
+; DISABLE-NEXT:    popq %rbx
+; DISABLE-NEXT:    .seh_endepilogue
+; DISABLE-NEXT:    retq
+; DISABLE-NEXT:  .LBB1_5: # %if.else
 ; DISABLE-NEXT:    addl %edx, %edx
 ; DISABLE-NEXT:    movl %edx, %eax
-; DISABLE-NEXT:  .LBB1_5: # %if.end
```

Comment:
Adding an SEH directive shouldn't affect block layout/merging this way.  Can we fix the heuristic here so it isn't sensitive to this?

Reply:
I did - the SEH instructions weren't marked as being meta-instructions, so when I first added the new instructions it broke some of the "taildup" tests.

After I marked them as meta-instructions, this test changed - presumably because it now sees the `jmp` is no longer profitable because it knows the SEH instructions won't produce real instructions (before my change `.seh_epilogue` was still being emitted where `.seh_startepilogue` now is, but it wasn't being streamed into the ASM listing).

Reply:
If it's just the addition of isMeta to seh_epilogue that's causing this, please separate that out into its own patch.